tests/resources/ddo_helpers.py

```diff
@@ -92,106 +90,184 @@ def get_access_service(
     )
 
 
-def get_registered_ddo(
-    ocean_instance,
-    metadata,
-    wallet: Wallet,
-    service=None,
-    datatoken=None,
-    provider_uri=None,
-):
-    metadata["main"]["files"][0]["checksum"] = str(uuid.uuid4())
-
-    if not service:
-        service = get_access_service(
-            ocean_instance,
-            wallet.address,
-            metadata["main"]["dateCreated"],
-            provider_uri,
-        )
-
-    block = ocean_instance.web3.eth.block_number
-    asset = ocean_instance.assets.create(
-        metadata,
-        wallet,
-        services=[service],
-        data_token_address=datatoken,
-        provider_uri=provider_uri,
-    )
-    ddo_reg = ocean_instance.assets.ddo_registry()
-    log = ddo_reg.get_event_log(
-        ddo_reg.EVENT_METADATA_CREATED, block, asset.asset_id, 30
+def create_asset(ocean, publisher, config, metadata=None):
+    """Helper function for asset creation based on ddo_sa_sample.json."""
+    erc20_data = ErcCreateData(
+        template_index=1,
+        strings=["Datatoken 1", "DT1"],
+        addresses=[
+            publisher.address,
+            publisher.address,
+            ZERO_ADDRESS,
+            get_address_of_type(config, "Ocean"),
+        ],
+        uints=[to_wei(100), 0],
+        bytess=[b""],
     )
-    assert log, "no ddo created event."
 
-    # Mint tokens for dataset and assign to publisher
-    dt = ocean_instance.get_data_token(asset.data_token_address)
-    mint_tokens_and_wait(dt, wallet.address, wallet)
+    if not metadata:
+        metadata = {
+            "created": "2020-11-15T12:27:48Z",
+            "updated": "2021-05-17T21:58:02Z",
+            "description": "Sample description",
+            "name": "Sample asset",
+            "type": "dataset",
+            "author": "OPF",
+            "license": "https://market.oceanprotocol.com/terms",
+        }
+    data_provider = DataServiceProvider
+    file1_dict = {"type": "url", "url": "https://url.com/file1.csv", "method": "GET"}
```

Comment:
I added the encrypt function on assets in my other branch (marketplace flow). If you think it helps here with the files, you can add it too

Reply:
Done.